src/tests/jsonLogicFixtures.js

```diff
@@ -0,0 +1,1098 @@
+  'x-jsf-logic': {
+    validations: {
+      require_c: {
+        rule: {
+          and: [
+            { '!==': [{ var: 'field_b' }, null] },
+            { '!==': [{ var: 'field_a' }, null] },
+            { '>': [{ var: 'field_a' }, { var: 'field_b' }] },
+          ],
+        },
+      },
+    },
+    allOf: [
+      {
+        if: {
+          validations: {
+            require_c: {
+              const: true,
+            },
+          },
+          properties: {
+            field_a: {
+              const: 10,
+            },
+          },
+        },
+        then: {
+          required: ['field_c'],
+        },
+        else: {
+          properties: {
+            field_c: false,
+          },
+        },
+      },
+    ],
+  },
+};
+
+export const schemaWithChecksAndThenValidationsOnThen = {
+  properties: {
+    field_a: {
+      type: 'number',
+    },
+    field_b: {
+      type: 'number',
+    },
+    field_c: {
+      type: 'number',
+    },
+  },
+  required: ['field_a', 'field_b'],
+  'x-jsf-logic': {
+    validations: {
+      c_must_be_large: {
+        errorMessage: 'Needs more numbers',
+        rule: {
+          '>': [{ var: 'field_c' }, 200],
+        },
+      },
+      require_c: {
+        rule: {
+          and: [
+            { '!==': [{ var: 'field_b' }, null] },
+            { '!==': [{ var: 'field_a' }, null] },
+            { '>': [{ var: 'field_a' }, { var: 'field_b' }] },
+          ],
+        },
+      },
+    },
+    allOf: [
+      {
+        if: {
+          validations: {
+            require_c: {
+              const: true,
+            },
+          },
+        },
+        then: {
+          required: ['field_c'],
+          properties: {
+            field_c: {
+              description: 'I am a description!',
+              'x-jsf-requiredValidations': ['c_must_be_large'],
+            },
+          },
+        },
+        else: {
+          properties: {
+            field_c: false,
+          },
+        },
+      },
+    ],
+  },
+};
+
+export const schemaWithComputedValueChecksInIf = {
+  properties: {
+    field_a: {
+      type: 'number',
+    },
+    field_b: {
+      type: 'number',
+    },
+    field_c: {
+      type: 'number',
+    },
+  },
+  required: ['field_a', 'field_b'],
+  'x-jsf-logic': {
+    computedValues: {
+      require_c: {
+        rule: {
+          and: [
+            { '!==': [{ var: 'field_b' }, null] },
+            { '!==': [{ var: 'field_a' }, null] },
+            { '>': [{ var: 'field_a' }, { var: 'field_b' }] },
+          ],
+        },
+      },
+    },
+    allOf: [
+      {
+        if: {
+          computedValues: {
+            require_c: {
+              const: true,
+            },
+          },
+        },
+        then: {
+          required: ['field_c'],
+        },
+        else: {
+          properties: {
+            field_c: false,
+          },
+        },
+      },
+    ],
+  },
+};
+
+export const schemaWithMultipleComputedValueChecks = {
```

Comment:
Interesting case. Here I'd expect the root field_c to have `'x-jsf-requiredValidations` declared even if in the conditional it it's disallowed (the `then ... false` bit). Which doesn't make sense as it's contradictory.

Perhaps a sanity test showing that impossible case would be useful for future reference.